src/test/java/org/opensearch/ad/e2e/DetectionResultEvalutationIT.java

```diff
@@ -197,20 +216,89 @@ private void startDetector(
         } while (true);
     }
 
-    private String createDetector(String datasetName, int intervalMinutes, RestClient client) throws Exception {
-        Request request = new Request("POST", "/_opendistro/_anomaly_detection/detectors/");
-        String requestBody = String
-            .format(
-                Locale.ROOT,
-                "{ \"name\": \"test\", \"description\": \"test\", \"time_field\": \"timestamp\""
-                    + ", \"indices\": [\"%s\"], \"feature_attributes\": [{ \"feature_name\": \"feature 1\", \"feature_enabled\": "
-                    + "\"true\", \"aggregation_query\": { \"Feature1\": { \"sum\": { \"field\": \"Feature1\" } } } }, { \"feature_name\""
-                    + ": \"feature 2\", \"feature_enabled\": \"true\", \"aggregation_query\": { \"Feature2\": { \"sum\": { \"field\": "
-                    + "\"Feature2\" } } } }], \"detection_interval\": { \"period\": { \"interval\": %d, \"unit\": \"Minutes\" } }, "
-                    + "\"schema_version\": 0 }",
-                datasetName,
-                intervalMinutes
-            );
+    /**
+     * Different from simulateSingleStreamStartDetector, getDetectionResult is an async process and we cannot use exception to decide if
```

Comment:
single stream detectors will throw exception if not finding models in the callback, while HCAD detectors will return early, record the exception in node state, and throw exception in the next run. HCAD did it this way since it does not know when current run is gonna finish (e.g, we may have millions of entities to process in one run). So for single-stream detector test case, we can check the exception to see if models are initialized or not. So HCAD, we have to either wait for next runs or use profile API. Here I chose profile API since it is faster. Will add these explanation in the comments.